$MirOS: ports/devel/glib/patches/patch-glib_gconvert_c,v 1.1 2006/06/04 20:57:15 bsiegert Exp $
--- glib/gconvert.c.orig	Mon Dec 18 04:20:16 2006
+++ glib/gconvert.c	Thu Mar  6 21:59:28 2008
@@ -41,13 +41,6 @@
 
 #include "glibintl.h"
 
-#if defined(USE_LIBICONV_GNU) && !defined (_LIBICONV_H)
-#error GNU libiconv in use but included iconv.h not from libiconv
-#endif
-#if !defined(USE_LIBICONV_GNU) && defined (_LIBICONV_H)
-#error GNU libiconv not in use but included iconv.h is from libiconv
-#endif
-
 #include "galias.h"
 
 GQuark 
@@ -728,6 +721,9 @@ g_convert (const gchar *str,
   g_return_val_if_fail (to_codeset != NULL, NULL);
   g_return_val_if_fail (from_codeset != NULL, NULL);
   
+  if (strcmp (to_codeset, from_codeset) == 0)
+    return g_strndup (str, len);
+
   cd = open_converter (to_codeset, from_codeset, error);
 
   if (cd == (GIConv) -1)
